nixos/tests/birdsitelive.nix

```diff
@@ -0,0 +1,118 @@
+      -subj '/CN=birdsitelive.nixos.test/' -days 49710 \
+      -addext 'subjectAltName = DNS:birdsitelive.nixos.test' \
+      -keyout "$out/key.pem" -newkey ed25519 \
+      -out "$out/cert.pem" -noenc
+  '';
+
+  check = pkgs.writers.writeBashBin "check" ''
+    set -e -u -o errtrace -o pipefail
+
+    paths=( / /About{,/{Blacklisting,Whitelisting}} )
+
+    for path in "''${paths[@]}"; do
+      ${pkgs.curl}/bin/curl -f -S -s -o /dev/null -w '%{response_code}' \
+        "https://birdsitelive.nixos.test$path" \
+        | grep -F -q 200
+    done
+  '';
+
+  hosts = nodes: ''
+    ${nodes.birdsitelive.networking.primaryIPAddress} birdsitelive.nixos.test
+    ${nodes.client.networking.primaryIPAddress} client.nixos.test
+  '';
+in
+{
+  name = "birdsitelive";
+  nodes = {
+    client = { nodes, pkgs, config, ... }: {
+      security.pki.certificateFiles = [ "${tlsCert}/cert.pem" ];
+      networking.extraHosts = hosts nodes;
+      environment.systemPackages = with pkgs; [ check ];
+    };
+
+    birdsitelive = { nodes, pkgs, config, ... }: {
+      security.pki.certificateFiles = [ "${tlsCert}/cert.pem" ];
+      networking.extraHosts = hosts nodes;
+      networking.firewall.allowedTCPPorts = [ 443 ];
+
+      services = {
+        postgresql = {
+          enable = true;
+          package = pkgs.postgresql_14;
+          initialScript = pkgs.writeText "init.sql" ''
+            CREATE ROLE birdsitelive
+              LOGIN PASSWORD '${dbPassword}';
+            CREATE DATABASE birdsitelive
+              OWNER birdsitelive
+              TEMPLATE template0
+              ENCODING 'utf8'
+              LOCALE 'C';
+          '';
+        };
+
+        birdsitelive = {
+          enable = true;
+          settings = {
+            Instance = {
+              Domain = "birdsitelive.nixos.test";
+              AdminEmail = "birdsitelive@nixos.test";
+            };
+            Db = {
+              Password = {
+                _secret = "${pkgs.writeText "dbPassword" dbPassword}";
+              };
+            };
+            Twitter = {
+              ConsumerKey = descramble "r35h4FJ1JlexgjNNtv3IONgfu";
+              ConsumerSecret = {
+                _secret = "/run/keys/twitter-api-key";
+              };
+            };
+          };
+        };
+
+        nginx = {
+          enable = true;
+          virtualHosts."birdsitelive.nixos.test" = {
+            addSSL = true;
+            sslCertificate = "${tlsCert}/cert.pem";
+            sslCertificateKey = "${tlsCert}/key.pem";
+
+            locations."/" = {
+              proxyPass = "http://localhost:5000";
+            };
+          };
+        };
+      };
+    };
+  };
+
+  testScript = { nodes, ... }: ''
+    start_all()
+    birdsitelive.succeed(('echo F2SkZ6NNhuNjgbRVPHbYsOsJXpxKlCJxzS5xhkkw9RXuKq4hLL'
+      ' | tr g5cd3Y1GroT9k42iPnK6sqHBfwSjuIVv0XAxtFNCQDUpLWzElORyZMaehJ8mb7'
+      ' 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz'
```

Comment:
Are those dummy or real credentials?

Reply:
> networking is for host ip networking. You want to move this under services

Moved it from `services/networking` to `services/web-apps` where most of the other ActivityPub implementations reside.

Reply:
> Are those dummy or real credentials?

These are real credentials. I requested them specifically for this test.

Obscuring them in this fashion does obviously not prevent them from being extracted, but this not my concern. If they are being (ab)used for something else, that would simply eat up the request quota for these credentials.

I would just like to prevent them from being detected by tools automatically scraping the web for them.

Reply:
Could you add a short comment with that information? Otherwise the next person could be confused as me.